### PR TITLE
feat: Add service_directory_registrations: namespace and service_directory_region

### PR DESCRIPTION
**Pull request by sanadhis**
Hi, I would propose to add optional variable of `project_region` for this module.

The idea is to allow specifying custom, not in us-central1, namespace link.

### CODE REVIEW

**Comment by imrannayer** [comment]
@sanadhis thx for the PR. Can you plz add two optional variables

[service_directory_region](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_global_forwarding_rule#service_directory_region)
[service_directory_namespace](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_global_forwarding_rule#namespace)

and use dynamic blocks if any of these values are provided.

File: modules/private-service-connect/variables.tf [hunk 1]
```diff
@@ -56,3 +56,15 @@ variable "forwarding_rule_target" {
     error_message = "For forwarding_rule_target only `all-apis` and `vpc-sc` are valid."
   }
 }
+
+variable "service_directory_namespace" {
```
Comment (imrannayer) [inline modules/private-service-connect/variables.tf:60]: Can we set this to null

**Comment by sanadhis** [comment]
Hi @imrannayer thanks for the feedback!

I've addressed them, can you re-review? Thanks 😄

**Comment by sanadhis** [comment]
Hi @imrannayer can we merge here if it's all okay? thanks 😄

**Comment by imrannayer** [comment]
@sanadhis can u plz follow [this doc](https://github.com/terraform-google-modules/terraform-google-network/blob/master/CONTRIBUTING.md) to fix the linit issues?

**Comment by sanadhis** [comment]
> @sanadhis can u plz follow [this doc](https://github.com/terraform-google-modules/terraform-google-network/blob/master/CONTRIBUTING.md) to fix the linit issues?

thanks for the hint.
Lint test passes in my machine now:
```
...
terraform_validate ./test/fixtures/simple_project_with_regional_network
Success! The configuration is valid.

terraform_validate ./test/fixtures/submodule_firewall
Success! The configuration is valid.

terraform_validate ./test/setup
Success! The configuration is valid.

ENABLE_BPMETADATA not set to 1. Skipping metadata validation.
```

**Comment by imrannayer** [comment]
@sanadhis this feature was part of [5.8](https://github.com/terraform-google-modules/terraform-google-network/blob/master/modules/private-service-connect/versions.tf). Can you plz update [version.tf](https://github.com/terraform-google-modules/terraform-google-network/blob/master/modules/private-service-connect/versions.tf) file to use provider version >= 5.8?

Thanks

File: docs/upgrading_to_v9.0.0.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Upgrading to v9.0.0
+
+The v9.0 release contains backwards-incompatible changes.
```
Comment (imrannayer) [inline docs/upgrading_to_v9.0.0.md:3]: @sanadhis next version will be 10.0

**Comment by sanadhis** [comment]
@imrannayer should I update `modules/private-service-connect/metadata.yaml` and provider version for `modules/private-service-connect/versions.tf` or do you have automation when releasing this module?

**Comment by imrannayer** [comment]
@sanadhis thats not needed. version meta will update by release scripts

**Review by imrannayer — APPROVED** [review]

**Comment by stanimprover** [comment]
@sanadhis @imrannayer I'm getting an Unsupported block type Error for the dynamic "service_directory_registrations" for the private-service-connect module and I have changed to multiple versions and still not working. Are you having this issue as well?

**Comment by imrannayer** [comment]
@stanimprover can u plz create an issue. This is part of 9.1 which is not released yet. Can you also post the code you are using for your testing?

**Pull request merged**